### PR TITLE
bug: [PagoPA-2093] - GPD error 500

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- added management for sending receipt-ko in case of exception
- added junit test

#### Motivation and Context

- the request was not closed in the event of an error and remained pending

#### How Has This Been Tested?

- Manual and junit test

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java [hunk 1]
```diff
@@ -30,36 +34,68 @@ public class ConverterService {
...
+    	                    .fiscalCode(sessionData.getCommonFields().getCreditorInstitutionId())
+    	                    .noticeNumber(noticeNumber)
+    	                    .build());
+    	            receiptService.sendKoPaaInviaRtToCreditorInstitution(inputPaaInviaRTKo);
+    			}
+    		}
+    		throw appException;   		
+    	} catch (Exception ex) {
+    		log.error("An error occurred during covert operations: " + ex.getMessage());
+    		// TODO decide what to do if it is not possible to recover the sessionData obj or if the information necessary to valorise the sendKoPaaInviaRtToCreditorInstitution is not present.
```
Comment (cap-ang) [inline src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java:85]: maybe the data to send a `paaInviaRT-` could be taken from the RPT (the NAV created by GPD should not be necessary for sending an RT alias `paaInviaRT`)

File: src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java [hunk 2]
```diff
@@ -30,36 +34,68 @@ public class ConverterService {
...
+    		if (sessionData != null && sessionData.getCommonFields() != null && !CollectionUtils.isEmpty(sessionData.getNAVs())) {
```
[the remark this answers is not in the document]
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java]: done

File: src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java]: improvement that will be discussed for future development

**Review by cap-ang — APPROVED** [review]
✅

File: src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java [hunk 3]
```diff
@@ -78,4 +95,18 @@ private void setECommerceHangTimer(SessionDataDTO sessionData) throws URISyntaxE
                         .noticeNumber(elem.getNoticeNumber())
                         .build()));
     }
+    
+    // TODO decide what to do if it is not possible to recover the sessionData obj or if the information necessary to valorise the sendKoPaaInviaRtToCreditorInstitution is not present.
+    private void sendKoPaaInviaRtToCreditorInstitution(SessionDataDTO sessionData) {
```
Comment (FedericoRuzzier) [inline src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java:100]: Refactor with new logic using only sessionId

File: src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java [hunk 4]
```diff
@@ -30,38 +34,51 @@ public class ConverterService {
...
-        return checkoutResponse;
+    public String convert(String sessionId) throws URISyntaxException {
+    	String checkoutResponse;
+    	SessionDataDTO sessionData = null;
+    	try {
+    		// get RPT request entity from database
+    		RPTRequestEntity rptRequestEntity = rptCosmosService.getRPTRequestEntity(sessionId);
+
+    		// unmarshalling and mapping RPT content from request entity, generating session data
+    		sessionData = this.rptExtractorService.extractSessionData(rptRequestEntity.getPrimitive(), rptRequestEntity.getPayload());
```
Comment (FedericoRuzzier) [inline src/main/java/it/gov/pagopa/wispconverter/service/ConverterService.java:49]: Exclude this method and the previous from try-catch, must be handled in a specific way

**Pull request merged**